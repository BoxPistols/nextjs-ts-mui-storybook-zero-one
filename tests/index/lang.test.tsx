import { render, screen } from '@testing-library/react'
import Home from 'pages/index'
// for with language test
import langJA from '../../locales/ja/lang.json'
import langEN from '../../locales/en/lang.json'
import langKO from '../../locales/ko/lang.json'
import I18nProvider from 'next-translate/I18nProvider'

describe('Home', () => {
  it('renders a heading', () => {
    const { container } = render(
      <>
        {/* JA */}
        <I18nProvider lang="ja" namespaces={{ lang: langJA }}>
          <Home />
        </I18nProvider>

        {/* EN */}
        <I18nProvider lang="en" namespaces={{ lang: langEN }}>
          <Home />
        </I18nProvider>

        {/* KOREA */}
        <I18nProvider lang="ko" namespaces={{ lang: langKO }}>
          <Home />
        </I18nProvider>
      </>,
    )
    expect(container).toMatchSnapshot()
  })
})
