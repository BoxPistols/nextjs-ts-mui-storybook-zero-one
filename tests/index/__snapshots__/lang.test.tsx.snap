// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a heading 1`] = `
<div>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Welcome to 
        <a
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <h2>
        日本語
      </h2>
      <h2>
        英語
      </h2>
      <h2>
        韓国語
      </h2>
    </main>
  </div>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Welcome to 
        <a
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <h2>
        Japanese
      </h2>
      <h2>
        English
      </h2>
      <h2>
        Korean
      </h2>
    </main>
  </div>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Welcome to 
        <a
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <h2>
        일본어
      </h2>
      <h2>
        영어
      </h2>
      <h2>
        한국어
      </h2>
    </main>
  </div>
</div>
`;
